nlmod/dims/layers.py

```diff
@@ -2269,6 +2266,50 @@ def get_max_overlap_model_layer(i):
     return modellayer
 
 
+def map_points_to_grid(x, y, ds=None, modelgrid=None):
+    """Map points to grid cells.
+
+    Parameters
+    ----------
+    x : np.ndarray
+        x-coordinates of points.
+    y : np.ndarray
+        y-coordinates of points.
+    ds : xr.Dataset, optional
+        Model dataset containing grid information. Must provide one of ds or modelgrid.
+    modelgrid : StructuredGrid or VertexGrid, optional
+        Model grid object. Must provide one of ds or modelgrid.
+
+    Returns
+    -------
+    cellids : pd.Series
+        series with mapping between points and cellid of grid cell in
+        which points are located.
+    """
+    if ds is None and modelgrid is None:
+        raise ValueError("Either ds or modelgrid must be provided.")
+    elif ds is not None:
+        modelgrid = grid.modelgrid_from_ds(ds)
+
+    # build geometries and cellids for grid
+    gi = flopy.utils.GridIntersect(modelgrid, method="vertex")
```

Comment:
and make gi an optional parameter, so users can supply their own GridIntersect (just like `nlmod.grid.get_icell2d_from_xy`)

Reply:
Done.